src/main/java/com/gmail/yuramitryahin/dao/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.gmail.yuramitryahin.dao;
+
+import com.gmail.yuramitryahin.exception.DataProcessingException;
+import com.gmail.yuramitryahin.lib.Dao;
+import com.gmail.yuramitryahin.model.MovieSession;
+import com.gmail.yuramitryahin.util.HibernateUtil;
+import java.time.LocalDate;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> getAllSessionsByDate =
+                    session.createQuery("SELECT ms FROM MovieSession ms "
+                                    + "LEFT JOIN FETCH ms.cinemaHall LEFT JOIN FETCH ms.movie"
```

Comment:
why are you using join fetch here? please answer in Slack

Reply:
fixed